app/models/site_statistic.rb

```diff
@@ -141,45 +197,179 @@ def self.observations_stats(at_time = Time.now)
       identified: identified,
       # community_identified: Observation.where("community_taxon_id IS NOT NULL AND created_at BETWEEN ? AND ?", at_time - 7.days, at_time).count,
       community_identified: community_identified,
-      community_identified_to_genus: Observation.joins(:community_taxon).
-        where("taxa.rank_level <= 20 AND observations.created_at BETWEEN ? AND ?", at_time - 7.days, at_time).
-        count
+      community_identified_to_genus: community_identified_to_genus
     }
   end
 
   def self.identifications_stats(at_time = Time.now)
     at_time = at_time.utc
+    count = Identification.elastic_search(
+      size: 0,
+      track_total_hits: true,
+      filters: [
+        range: {
+          created_at: {
+            lte: at_time
+          }
+        }
+      ]
+    ).total_entries
+    last_7_days = Identification.elastic_search(
+      size: 0,
+      track_total_hits: true,
+      filters: [
+        range: {
+          created_at: {
+            gte: at_time - 7.days,
+            lte: at_time
+          }
+        }
+      ]
+    ).total_entries
+    today = Identification.elastic_search(
+      size: 0,
+      track_total_hits: true,
+      filters: [
+        range: {
+          created_at: {
+            gte: at_time - 1.days,
+            lte: at_time
+          }
+        }
+      ]
+    ).total_entries
     {
-      count: Identification.where("created_at <= ?", at_time).count,
-      last_7_days: Identification.where("created_at BETWEEN ? AND ?", at_time - 7.days, at_time).count,
-      today: Identification.where("created_at BETWEEN ? AND ?", at_time - 1.day, at_time).count
+      count: count,
+      last_7_days: last_7_days,
+      today: today
     }
   end
 
   def self.users_stats(at_time = Time.now)
     at_time = at_time.utc
-    es_response = Observation.elastic_search(
+    count = User.elastic_search(
       size: 0,
       track_total_hits: true,
       filters: [
-        { terms: { quality_grade: %w(research needs_id) } },
         {
           range: {
             created_at: {
-              gte: ( at_time - 1 .day ),
+              lte: at_time
+            }
+          }
+        },
+        { term: { spam: false } }
+      ]
+    ).total_entries
+    curators = User.elastic_search(
+      size: 0,
+      track_total_hits: true,
+      filters: [
+        {
+          range: {
+            created_at: {
+              lte: at_time
+            }
+          }
+        },
+        { term: { spam: false } },
+        { term: { roles: "curator" } }
```

Comment:
To match the original count we should include the admin role as well. So we should make this a `terms` filter that includes both curator and admin

Reply:
Done